src/broadcast.jl

```diff
@@ -25,7 +25,7 @@ if VERSION < v"0.6.0-dev" # Old approach needed for inference to work
     @inline ziptype(A, B, C, D...) = Zip{Tuple{eltype(A)}, ziptype(B, C, D...)}
 
     nullable_broadcast_eltype(f, As...) =
-        _default_eltype(Base.Generator{ziptype(As...), ftype(f, As...)})
+        eltype(_default_eltype(Base.Generator{ziptype(As...), ftype(f, As...)}))
```

Comment:
No, `eltype(NullableArray([1], [true])) == Nullable{Int}`.

I don't want `T = Nullable{Int}` here since it's used to build a new array via `NullableArray{T}(...)`. `NullableArray{Nullable{Int}}` would have a `Nullable{Nullable{Int}}` eltype.